### PR TITLE
style(colors): restyle buttons, links and labels to conform to mozill…

### LINKED ISSUE #3754: Restyle buttons to conform to the Mozilla style guide
From IRC:
The style guide from shorlander has a state defined for the web:
http://people.mozilla.org/~shorlander/styleguide/ui-elements/buttons-all.html

We should ensure that FxA sticks to the style guide.

@ryanfeeley 
@shane-tomlinson 
@rfk

### PR DESCRIPTION
**Pull request by TDA**
…a style guide

edit primary and secondary buttons
edit show-password label
edit links
Fixes #3754

@ryanfeeley 
@rfk 
@shane-tomlinson

### CODE REVIEW

File: app/styles/_variables.scss [hunk 1]
```diff
@@ -2,24 +2,64 @@
...
+  --border-radius-osx: 4px;
+  --border-radius-outer: 4px;
+  --border-radius-inner: 3px;
+}
+
 //Color Variables
-$button-background-color: #0095dd;
-$button-background-disabled-color: #8a9ba8;
-$button-background-hover-color: #08c;
+$button-background-color: var(--color-blue);
```
Comment (shane-tomlinson) [inline app/styles/_variables.scss]: Will SASS convert this to the actual value? If not, ixnay, as CSS variables are only supported in very recent browsers, and still not in all recent browsers: http://caniuse.com/#feat=css-variables

Comment (TDA) [inline app/styles/_variables.scss]: @shane-tomlinson No it doesn't. So I was following the moz-style-guide by @shorlander here: http://people.mozilla.org/~shorlander/styleguide/ui-elements/buttons-all.html and it was using css variables. I could definitely just use sass variables instead. Should I do that?

**Comment by TDA** [comment]
@shane-tomlinson Updated with sass variables. Would benefit from cleanup as specified in #3753.

**Comment by TDA** [comment]
@shane-tomlinson updated with changes to input fields as well.

**Comment by shane-tomlinson** [comment]
@ryanfeeley - mind doing a UX review?

I'll note that the focus ring looks slightly different between input fields and buttons. The button looks like it matches the style guide. It looks like the input field focus ring has no inner padding.

I also suggest we should update the checkbox to have a blue border when focused.

### Style guide
screen shot 2016-05-23 at 10 57 22

### Input box
screen shot 2016-05-23 at 10 57 39

### Button
screen shot 2016-05-23 at 10 57 44

Here it looks like the button text is 1px too low too.

### Focused checkbox
screen shot 2016-05-23 at 11 02 31

------------------------

The settings page needs some work:

The secondary buttons are grey, whereas the style guide makes it seem like those should be white.

### Grey secondary buttons
screen shot 2016-05-23 at 11 07 59

### Missing text on delete account

The delete account text is missing.

delete-account

### Change password looks a bit funny.

Maybe it's the gray secondary button with the white disabled primary button.

screen shot 2016-05-23 at 11 08 41

**Comment by TDA** [comment]
@shane-tomlinson The delete account button on the settings page looks like this to me:
screenshot 2016-05-23 11 50 57

**Comment by ryanfeeley** [comment]
@shane-tomlinson 

> The focus ring has no inner margin/padding.

This matches the style guide. Make sure your zoom is 100%.

> The button text looks 1px too low.

It does. Maybe we don't need to specify button height? Seems to fix the issue when I tried it.

> The secondary buttons are grey, whereas the style guide makes it seem like those should be white.

They are closer to white, but not quite. Good eye though, our secondary buttons need to be lighter.

**Comment by TDA** [comment]
@ryanfeeley 
@shane-tomlinson 
Fixed the secondary button color, but I dont understand the part about button height/ font size, I didnt touch either of those :O

**Comment by shane-tomlinson** [comment]
> EDIT: Do you mean that the font size got affected due to box-sizing and me setting the border? If so, what should I do? Increase font-size? or button height?

Font size is the same AFAICT, so that's not the problem. Rather, alignment is. I don't know the cause, you'll have to figure out the why. Keep the font size, fix the button height or the text center alignment.

**Comment by TDA** [comment]
@shane-tomlinson OHHHHH, 1px too `low` :D :D hahaha, I thought you meant 1px too `small`, and hence the font-size question, my bad :D :D

**Comment by TDA** [comment]
@shane-tomlinson Fixed the button height to make sure the text stuck to the same box-model as before. Width reduction of 2px doesn't affect alignment :) so leaving as-is.
### Before

screenshot 2016-05-26 09 39 26

### After
screenshot 2016-05-26 09 39 10

**Comment by shane-tomlinson** [comment]
@ryanfeeley - Is the background color supposed to have changed? It looks a bit washed out now, especially with the chunky drop shadow:

### previous background color
screen shot 2016-05-27 at 14 20 04

### new background color

screen shot 2016-05-27 at 14 19 59

The drop shadow looks especially chunky on bottom and right of the the settings page main panel:

screen shot 2016-05-27 at 14 53 47

@ryanfeeley - what do you think if we use a similar style to activity stream, instead of a drop shadow around the box, we use a 1px border colored rgba(0,0,0,0.1)

### proposed border change

screen shot 2016-05-27 at 14 31 01

### SASS

```sass
diff --git a/app/styles/_layout.scss b/app/styles/_layout.scss
index 8991e33..ba5d137 100644
--- a/app/styles/_layout.scss
+++ b/app/styles/_layout.scss
@@ -18,7 +18,7 @@
 
   @include respond-to('big') {
     border-radius: $big-border-radius;
-    box-shadow: 0 1px 5px $input-row-border-color;
+    border: 1px solid rgba(0,0,0,0.1);
     margin: -15px auto 0 auto;
     min-height: 420px;
     padding: 50px 40px 40px 40px;
```

That's not a complete change, you can see the border on the bottom is still too large, and this does not handle the settings page.

## The focus ring is different between the input boxes and buttons of all sorts

### input box

screen shot 2016-05-27 at 14 48 01


### button

screen shot 2016-05-27 at 14 48 08

### Secondary button on the settings page

screen shot 2016-05-27 at 15 01 12

### Proposal 1 - Change the input focus ring

Here's one proposal where focus rings act more like the buttons

screen shot 2016-05-27 at 14 51 07

screen shot 2016-05-27 at 14 51 17

### SASS
The focus ring changes use two drop shadows. This may require other changes to hover/focus states, I haven't worked through it all, but here's the SASS I used:

```sass
diff --git a/app/styles/modules/_input-row.scss b/app/styles/modules/_input-row.scss
index 5a15377..b2309a5 100644
--- a/app/styles/modules/_input-row.scss
+++ b/app/styles/modules/_input-row.scss
@@ -50,13 +50,12 @@
       }
 
       &:focus {
-        border-color: $input-row-focus-border-color;
-        box-shadow: 0 0 0 2px $input-row-focus-box-shadow-color;
+        box-shadow: 0 0 0 1px $html-background-color, 0 0 0 3px $input-row-focus-box-shadow-color;
       }
 
       &.invalid {
         border-color: $error-background-color;
-        box-shadow: 0 0 0 2px rgba(lighten($error-background-color, 15), .75);
+        box-shadow: 0 0 0 1px $html-background-color, 0 0 0 3px rgba(lighten($error-background-color, 15), .75);
       }
     }
```

### Proposal 2 - Change the button focus ring

Here's a proposal where the buttons act more like the input elements

screen shot 2016-05-27 at 15 12 04

screen shot 2016-05-27 at 15 11 41

#### SASS

```sass
diff --git a/app/styles/modules/_button-row.scss b/app/styles/modules/_button-row.scss
index 8a47227..680eb99 100644
--- a/app/styles/modules/_button-row.scss
+++ b/app/styles/modules/_button-row.scss
@@ -55,7 +55,7 @@
     }
 
     &:focus {
-      border-color: $html-background-color;
       box-shadow: 0 0 0 2px $button-box-shadow-color;
     }
   }
```

I don't think any of my SASS is ready for prime time, but it's a couple ideas of how we could clean up a few things.

**Comment by TDA** [comment]
@shane-tomlinson Not sure about the background/box-shadow/border thing, but the buttons and the input are _supposed_ to be different, according to https://people.mozilla.org/~shorlander/styleguide/ui-elements/buttons.html and https://people.mozilla.org/~shorlander/styleguide/ui-elements/fields.html

@ryanfeeley thoughts? :)

**Comment by shane-tomlinson** [comment]
Thanks for the reference @TDA - that inconsistency is making my OCD kick in big time.

**Comment by ryanfeeley** [comment]
@shane-tomlinson It's not inconsistent if they are different things. A button is not a field, and having them appear differently helps the user distinguish between them.

**Comment by TDA** [comment]
@shane-tomlinson and @ryanfeeley Fixed the background. r?

**Comment by ryanfeeley** [comment]
👍

**Comment by shane-tomlinson** [comment]
@TDA - I was going to merge after fixing the sass errors, did some more testing, and found an issue that makes me uncomfortable. The settings page no longer shows active states when using the keyboard, which makes keyboard use nearly impossible. Screen readers can probably handle this, but other keyboard users that rely on visual queues will be missing important information.

### prod
button-focus-prod

### with this PR

button-focus-current

@ryanfeeley - what is your professional guidance?

**Comment by ryanfeeley** [comment]
@shane-tomlinson Can we apply the style guide focus to these buttons too? https://people.mozilla.org/~shorlander/styleguide/ui-elements/buttons-all.html

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson Can we apply the style guide focus to these buttons too? https://people.mozilla.org/~shorlander/styleguide/ui-elements/buttons-all.html

@TDA - This is so close! Can you make the changes @ryanfeeley suggests for the keyboard focus issue I outlined above?

**Comment by TDA** [comment]
@shane-tomlinson Fixed that. What I dont understand is why the styles from the button elements :focus didnt cascade :(
P.S: Also, I am fixing *every* sass related stuff in the sass refactor PR here: #3783 , including variable names and colorguard-audit. So lets leave the sass-related warnings to that PR :)

**Comment by shane-tomlinson** [comment]
>  So lets leave the sass-related warnings to that PR :) 

This doesn't jive with my OCD. Since we've added the lint tool, let's use it and not merge code we know will cause the lint tool to warn.

> What I dont understand is why the styles from the button elements :focus didnt cascade :(

You changed the order of the pseudo-selectors, that could possibly cause the side effects seen. 
The previous order was active, hover, focus, it is now hover active focus. What is the reason for the order change?

**Comment by shane-tomlinson** [comment]
@ryanfeeley - will email buttons need to be updated after this to match the colors?

### New primary button in content server

screen shot 2016-06-02 at 13 14 09

### Email button

screen shot 2016-06-02 at 13 14 24

**Comment by TDA** [comment]
@shane-tomlinson hover-active-focus _is_ the right ordering, if it were active-hover-focus, the hover state would _always_ override the active, thereby never giving us an active state.

**Comment by TDA** [comment]
@shane-tomlinson Corrected the sasslint-warnings. One more r please?

**Pull request merged**